### DISCUSSION TITLE
retry with backoff

### DISCUSSION THREAD

**Discussion by shibuiwilliam** [encode/httpx#1895] [Ideas] [answered]
Is there any way to do retry with backoff?
Retry with jitter backoff or exponential backoff might be good as well.

**Comment by mjuopperi** [comment, marked as answer]
Looks like that is built in. Looking at the source `httpx` uses `httpcore` for connections and exponential backoff is used there:

https://github.com/encode/httpcore/blob/e7a7b1924d2e47f2663361f531973eb3ee7d2ca9/httpcore/_async/connection.py#L157

**Reply by shibuiwilliam** [reply]
Thanks!
